factorial1.cpp

```diff
@@ -0,0 +1,20 @@
+
+#include <iostream>
+
+int main() 
+{
+  int fact = 1;
+  unsigned long long int N{};
+  std::cin >> N;
+  
+  if (N == 0 || N == 1)
+  fact = 1;
+  
+  else
```

Comment:
there is no need to separate N==0 and N==1 cases

Reply:
ok, I just used the math definition